pkg/apis/core/v1alpha1/webhook.go

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright 2020 VMWare, Inc.  All rights reserved. -- VMWare Confidential
+ *******************************************************************************/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// WebhookImpl implements webhook validator of a resource.
+type WebhookImpl interface {
+	Default(in *ExternalEntity)
+	ValidateCreate(in *ExternalEntity) error
+	ValidateUpdate(in *ExternalEntity, old runtime.Object) error
+	ValidateDelete(in *ExternalEntity) error
+}
+
+var (
+	externalEntityWebhook WebhookImpl
+)
+
+// RegisterWebhook registers webhook implementation of a resource.
+func RegisterWebhook(in runtime.Object, webhook WebhookImpl) error {
```

Comment:
I'm not familiar with webhook code, but where is this function called? I didn't see its usage in this PR and haven't seen other CRD based projects defining this. Do we follow some convention or examples?

Reply:
The quirky thing about ExternalEntities is that, it is defined in, and consumed by antrea, but it is not produced by antrea. 

In this case, the producer is Antrea-plus, would like to do some Webhooks processing.
The method is provided so that it may be called within sigs.k8s.io/controller-runtime (used as part of Kube-builder). Ideally we should have implements the Webhooks at where Webhooks service is provided, however Golang does not allow extend methods for a struct (ExternalEntity), hence the hook method to sigs.k8s.io/controller-runtime is defined here.

Reply:
Do you think we can remove this file for now until we are clear of the usage of ExternalEntity and what we want to put in Antrea repo?

Reply:
@jianjuns as discussed offline, we will need ExternalEntity Webhooks methods to validate.